packages/eslint-plugin-pf-codemods/lib/rules/v5/td-TdSelectType-TdActionsType-rename-disable.js

```diff
@@ -0,0 +1,92 @@
+const { getPackageImports } = require("../../helpers");
+
+// https://github.com/patternfly/patternfly-react/pull/8861
+// https://github.com/patternfly/patternfly-react/pull/8904
+module.exports = {
+  meta: { fixable: "code" },
+  create: function (context) {
+    const tableTdImport = getPackageImports(
+      context,
+      "@patternfly/react-table"
+    ).find((specifier) => "Td" === specifier.imported.name);
+
+    const tdName = tableTdImport?.local.name;
+
+    return !tableTdImport
+      ? {}
+      : {
+          JSXOpeningElement(node) {
+            if (tdName !== node.name?.name) {
+              return;
+            }
+
+            const attributesToUpdate = node.attributes.filter(
+              (attr) =>
+                attr.type === "JSXAttribute" &&
+                attr.value.type === "JSXExpressionContainer" &&
+                ["select", "actions"].includes(attr.name?.name)
+            );
+
+            const interfaceMap = {
+              select: "TdSelectType",
+              actions: "TdActionsType",
+            };
+
+            for (const attr of attributesToUpdate) {
+              const expr = attr.value.expression;
+
+              const getObjectProp = (expr, propName) =>
+                expr.properties.find(
+                  (prop) =>
+                    (prop.key.type === "Identifier" &&
+                      prop.key.name === propName) ||
+                    (prop.key.type === "Literal" && prop.key.value === propName)
+                );
+
+              const doReport = (disableProp) => {
+                if (disableProp) {
+                  context.report({
+                    message: `'disable' prop of interface ${
+                      interfaceMap[attr.name.name]
+                    } has been renamed to 'isDisabled'`,
+                    node,
+                    fix: function (fixer) {
+                      return fixer.replaceText(disableProp.key, "isDisabled");
+                    },
+                  });
+                }
+              };
+
+              if (expr.type === "ObjectExpression") {
+                doReport(getObjectProp(expr, "disable"));
+              }
+
+              if (expr.type === "Identifier") {
+                let scope = context.getScope();
+                while (scope !== null) {
+                  const variable = scope.variables.find(
+                    (variable) => variable.name === expr.name
+                  );
+                  if (!variable) {
+                    scope = scope.upper;
+                    continue;
+                  }
+                  if (
+                    variable.references.some(
+                      (ref) => ref.identifier.loc === expr.loc
+                    )
+                  ) {
+                    variable.defs.forEach((def) => {
```

Comment:
This is more me trying to understand the structure of the variable interface... but would there ever be more than a single definition for a variable?

btw, I tried doing 

      import { Td } from "@patternfly/react-table";
      const isDisabled = true;
      let myObj = {"disable": isDisabled};
      myObj = { "disable": false};
      <Td select={myObj} actions={{disable: false}} ></Td>`,

and it converted to

            import { Td } from "@patternfly/react-table";
            const isDisabled = true;
            let myObj = {isDisabled: isDisabled};
            myObj = { "disable": false};
             <Td select={myObj} actions={{isDisabled: false}} ></Td>

note that it correctly updated the initialization but not the update, this is not a breaker for me as it's probably an edge case that nobody would be using.. but if you wanted to look into it

Reply:
About the _would there ever be more than a single definition for a variable?_ It may be a case if we would ignore scopes, but each variable has its scope, so I am also confused, why is the `Variable.defs` property an array.